client/state/jetpack-connect/reducer/jetpack-connect-authorize.js

```diff
@@ -65,35 +54,43 @@ export default function jetpackConnectAuthorize( state = {}, action ) {
 				authorizeSuccess: false,
 				autoAuthorize: false,
 			} );
+
 		case JETPACK_CONNECT_AUTHORIZE_LOGIN_COMPLETE:
 			return Object.assign( {}, state, { authorizationCode: action.data.code } );
+
 		case JETPACK_CONNECT_AUTHORIZE_RECEIVE_SITE_LIST:
-			const updateQueryObject = omit( state.queryObject, '_wp_nonce', 'secret', 'scope' );
-			return Object.assign( {}, omit( state, 'queryObject' ), {
+			return Object.assign( {}, state, {
 				siteReceived: true,
 				isAuthorizing: false,
-				queryObject: updateQueryObject,
+				queryObject: omit( state.queryObject, '_wp_nonce', 'secret', 'scope' ),
 			} );
+
 		case JETPACK_CONNECT_QUERY_SET:
-			const queryObject = Object.assign( {}, action.queryObject );
 			const shouldAutoAuthorize = includes(
 				[ 'woocommerce-services-auto-authorize', 'woocommerce-setup-wizard' ],
-				queryObject.from
+				action.queryObject.from
 			);
-			const additionalQueryArgs = shouldAutoAuthorize ? { autoAuthorize: true } : {};
 			return Object.assign(
-				{},
-				buildDefaultAuthorizeState(),
-				{ queryObject },
-				additionalQueryArgs
+				{
+					queryObject: action.queryObject,
```

Comment:
Any reason this needed to use a "clone" as before? I don't see why that would be important…